share/wake/lib/system/remote_cache_api.wake

```diff
@@ -533,78 +533,99 @@ export target rscApiCheckAuthorization (api: RemoteCacheApi): Result Unit Error
+        "db" -> dbBlobFn
+        "file" -> dbFileFn
+        "http" -> dbFileFn
+        "https" -> dbFileFn
+        _ -> (\s unsupportedScheme s)
+
+    def uriFn (uri: String): Result a Error =
+        def parts = extract `([a-zA-Z][a-zA-Z0-9+.-]*)://(.*)` uri
+
+        match parts
+            scheme, path, Nil -> (schemeFn scheme) scheme path
+            _ -> failWithError "rsc: uri has unexpected format: '{uri}'"
+
+    uriFn
+
 # rscApiGetStringBlob: Downloads a blob and returns the contents as a string 
 #
 # ```
 #  rscApiGetStringBlob (RemoteCacheBlob "asdf" "https://...") = Pass "foo\nbar\nbat" 
 # ```
 export def rscApiGetStringBlob ((CacheSearchBlob _ uri): CacheSearchBlob): Result String Error =
-    require scheme, path, Nil = extract `([a-zA-Z][a-zA-Z0-9+.-]*)://(.*)` uri
-    else failWithError "rsc: uri has unexpected format: '{uri}'"
-
-    # maps path = "%46%6F%6F" to content = "foo"
-    def dbScheme _scheme path =
-        # If path is empty we have the empty string
-        require True = path.strlen > 0
-        else Pass ""
-
-        # If not empty then it must as least be %00
-        require True = path.strlen >= 3
-        else failWithError "rsc: Invalid db path: '{path}'"
-
-        require "", byteStrs = tokenize `%` path
-        else failWithError "rsc: Failed to extract bytes from db path: '{path}'"
-
-        require Some bytes =
-            byteStrs
-            | map (intbase 16)
-            | findNone
-        else failWithError "rsc: Failed to parse bytes strings into bytes from db path: '{path}'"
-
-        bytes
-        | map integerToByte
-        | cat
-        | Pass
-
-    def fileScheme scheme path =
+    def dbFileScheme scheme path =
         require Pass response =
             buildHttpRequest "{scheme}://{path}"
             | setMethod HttpMethodGet
             | makeRequest
 
         Pass response.getHttpResponseBody
 
-    def unsupportedScheme scheme path =
-        failWithError "rsc: scheme '{scheme}' from uri '{uri}' is not supported"
-
-    def schemeFn = match scheme
-        "db" -> dbScheme
-        "file" -> fileScheme
-        "http" -> fileScheme
-        "https" -> fileScheme
-        _ -> unsupportedScheme
+    def dbBlobScheme _scheme path = rscDbBlobToContent path
+    def schemeFn = uriSchemeFn dbBlobScheme dbFileScheme
 
-    schemeFn scheme path
+    schemeFn uri
 
 # rscApiGetFileBlob: Downloads a blob to *path* with *mode* permisssions and return *path*
 #
 # ```
 #  rscApiGetFileBlob (RemoteCacheBlob "asdf" "https://...") "foo/bar" 0644 = Pass "foo/bar" 
 # ```
 export def rscApiGetFileBlob ((CacheSearchBlob _ uri): CacheSearchBlob) (path: String) (mode: Integer): Result String Error =
-    require Pass downloadPath =
-        buildHttpRequest uri
+    def dbBlobScheme _scheme uripath =
+        require Pass content = rscDbBlobToContent uripath
+
+        writeTempFile "rsc.output_file.blob" content
+
+    def dbFileScheme scheme uripath =
```

Comment:
`content` here is the raw bytes from the return? Do we know for sure the bytes are valid for being in a linux path name? Would prefer if we hash it just to be safe

Reply:
nvm I'm just tired, hashing is done for you